src/diffpy/utils/scattering_objects/diffraction_objects.py

```diff
@@ -763,25 +763,27 @@ def q_to_tth(self):
 
             2\theta_n = 2 \arcsin\left(\frac{\lambda q}{4 \pi}\right)
 
+        Function adapted from scikit-beam.  Thanks to those developers
+
         Parameters
         ----------
         q : array
-            An array of :math:`q` values
+            The array of :math:`q` values
 
         wavelength : float
             Wavelength of the incoming x-rays
 
-        Function adapted from scikit-beam.  Thanks to those developers
-
         Returns
         -------
         two_theta : array
-            An array of :math:`2\theta` values in radians
+            The array of :math:`2\theta` values in radians
         """
         q = self.on_q[0]
         q = np.asarray(q)
         wavelength = float(self.wavelength)
         pre_factor = wavelength / (4 * np.pi)
+        if np.any(np.abs(q * pre_factor) > 1):
+            raise ValueError("Please check if you entered an incorrect wavelength or q value.")
```

Comment:
I think we would want to raise a value error here because otherwise the code proceeds with tth = nan.